sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -363,6 +364,70 @@ trait Row extends Serializable {
     false
   }
 
+  protected def canEqual(other: Any) =
+    other.isInstanceOf[Row] && !other.isInstanceOf[InternalRow]
```

Comment:
cc @chenghao-intel , I think `other` won't be null as we won't compare row and null in our code, right?
And it does't make sense to compare a external row and a internal row, but we should allow compare different kind of internal rows, so `other.getClass == this.getClass` may not be a good idea?